### DISCUSSION TITLE
A question about the license

### DISCUSSION THREAD

**Discussion by ChattiAchrafTn** [ultralytics/ultralytics#1260] [General]
I tried to read the license and the usage of many legal confused me.
English is not my first language so can someone explain is my case free to use the yolov8 models and modifying them without paying ?

I am a student in an internship that needs to make a project for a medium company I decided to use yolov8 since it's the latest and fastest with a chance to modify some of the hyperparameters 

does the company need to buy the "Enterprise License" once its start working with the solution based on yolov8 ?

**Comment by AyushExel** [comment]
Hey, the license is not meant to hamper any development it's just a safeguard against unethical use cases. Please feel free to continue using it however you want and leave it up to your company to determine if they need it. We'll be happy to assist them further if you raise a request here -> https://ultralytics.com/license

**Comment by SpirikleOfficial** [comment]


**Reply by pderrenger** [reply]
Hi there! 👋 Thanks for sharing your experience. We understand that the licensing fee might seem high for startups, especially those just beginning their journey. Our licensing fees are structured to support the ongoing development and maintenance of the YOLO models, ensuring they remain state-of-the-art. 

We encourage startups and individuals with budget constraints to explore our AGPL-3.0 licensed models, which can be used under the open-source license conditions. This might be a viable option if you're able to comply with the AGPL-3.0 requirements. 

For specific projects or further clarification on licensing, feel free to reach out to us directly. We're here to help and support the community in deploying cutting-edge AI solutions. 🚀

**Comment by jadrian21** [comment]
Hi, I'm working on a project that will Analyse up to **200 video feeds** on One Site (NDA so cannot explain) what pricing would I expect?
Just an Enterprise license per site? ($5000/Year) or if I set up more sites (Could be up to 20+). When would I be expected to pay the
License? When I have sold the project to the client commercially? I'm using the Open Source for research and development the end
product cannot be Open Source (Commercial confidentiality). Please answer this request as this project could go eventually world-wide
and be very big (1000's of sites). Thank you.

**Reply by jadrian21** [reply]
Just to add someone mentioned "Hey, the license is not meant to hamper any development it's just a safeguard against unethical use cases."
can you give examples of unethical use cases, we are not making any weapons systems or pornography software?? Or is it more subtle.
Our system involves the Medical Field (e.g. In Surgery log medical equipment usage. **This is not what we're doing it's just an example**)

**Reply by pderrenger** [reply]
@jadrian21 hello! Yes, the aim of the license is indeed to protect against unethical use cases while encouraging innovative and positive applications of YOLOv8. Examples of unethical use might include applications that infringe on privacy rights, illegal surveillance, or any technology that could cause harm to individuals or groups. Your medical field example, such as logging medical equipment usage in surgery (noted it's just an example), sounds like a beneficial and ethical application. The key is to ensure that technology enhances and safeguards human dignity and rights. If you have any more specific questions or need further clarification, feel free to ask! 😊

**Reply by jadrian21** [reply]
The reason I used this example is that the video feeds may need to look at the whole surgery theatre including the surgeons, nurses, and patient to track the instruments to log usage and prevent left behind/misplaced medical equipment. It could alert left behind equipment in situ for example.
The only people looking at the feeds for training purposes would be qualified medical professionals. No actual video footage would be
stored when in production. Again I stress this is an example.

Also, can someone answer the original question about licensing thank you.

**Reply by pderrenger** [reply]
@jadrian21 hello! Your example application of using YOLOv8 to track medical instruments in a surgery theatre to improve safety and efficiency sounds innovative and ethical. Ensuring that only qualified medical professionals are involved in the training process and not storing actual video footage in production aligns well with privacy and ethical considerations. 👍

Regarding your original question about licensing: For your internship project at a medium company where you're considering modifying YOLOv8's hyperparameters and deploying the solution, an **Enterprise License** would indeed be required once the project transitions from an R&D stage to being used commercially within the company. The AGPL-3.0 license allows for modification and use in open source projects, but if the end solution is proprietary and used internally by a company or commercially, an Enterprise License is necessary.

Feel free to reach out if you have more questions or need further clarification!

**Comment by openstackerexplorer** [comment]
hi, I have one query regarding licence. If the application is being used in an company but not for commercial purpose/gain but more in R&D setup. Is there a need to procure licence for it ?

**Reply by glenn-jocher** [reply]
@AbhiRathore hello,

Any usage of Ultralytics models, including in an R&D setup within a company, requires an Ultralytics Enterprise License unless the entire project is open-sourced under the AGPL-3.0 license. This applies regardless of whether the usage is commercial or not. For more details or to apply for an Enterprise License, please visit https://ultralytics.com/license.

**Comment by stefp** [comment]
Hi, I am also a bit confused about licencing. Here is my case:
I am a forestry researcher and I trained a YOLOv11 model with my own data, I plan to publish my code in a gihub repo along a paper. In addition we have developed our own cloud service where external users (both researchers and companies) will run the inference using my trained model. Is this case falling under the AGPL-3.0 license given that I publish my code? If so is there any specific paperwork I should do to comply with this licence or is it enough to publish the code under the same licence?

Thanks in advance

**Reply by glenn-jocher** [reply]
@stefp if you publish your code under the AGPL-3.0 license, your use case should comply with the license requirements. Ensure your entire project is open-sourced under the same license. For specific legal advice, consulting a legal professional is recommended.

**Reply by stefp** [reply]
thanks a lot @glenn-jocher for the clarification

**Reply by glenn-jocher** [reply]
You're welcome! If you have any more questions, feel free to ask.

**Comment by jslok** [comment]
@glenn-jocher What is the cost of a licence for startup/one man devs? I sent several messages and didn't get a reply for weeks. Then I got a reply just telling me to reply back for more info. Haven't heard back since. Do you only sell licenses to big businesses? What's the minimum?

**Reply by glenn-jocher** [reply]
@jslok thank you for your interest! Ultralytics licenses are tailored to the specific needs of each user, regardless of company size. Please fill out the [license request form](https://ultralytics.com/license), and our team will provide a quote and further details.

**Comment by Parth-cilans** [comment]
Hey
I am also a bit confused about licensing. Here is my case: If I fine-tune YOLOv8 on my own dataset and use it internally within my company, for example, to detect vehicle license plates in our company’s parking lot, but do not sell, distribute, or offer this system as a commercial product or service, would I still be required to purchase an enterprise license?

The system will only be used within the company for proprietary purposes and will not be monetized in any way. I want to ensure compliance with AGPL-3.0 and understand if an enterprise license is necessary in this case.

**Reply by glenn-jocher** [reply]
@Parth-cilans yes, any internal company use of Ultralytics YOLO models (including modified YOLOv8) requires either an Enterprise License or open-sourcing your entire project under AGPL-3.0. Since this is proprietary internal usage without public release of source code and trained models, your company would need an Enterprise License. For full compliance details, visit Ultralytics Licensing.

**Comment by emitoooo** [comment]
Hello, I have a question about license. If I use Ultralytics models for my company for multiple use cases, how do you calcualte the pricing? Is it calculated per training, or deployed projects or number of fine-tuned models? Or it is one time fee per year?
